lib/mobility-core/src/Kernel/External/Notification/PayTM/Client.hs

```diff
@@ -0,0 +1,48 @@
+{-
+  Copyright 2022-23, Juspay India Pvt Ltd
+
+  This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+  as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program is
+
+  distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+
+  FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of the GNU Affero
+
+  General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Kernel.External.Notification.PayTM.Client where
+
+import EulerHS.Prelude
+import Kernel.External.Encryption (decrypt)
+import qualified Kernel.External.Notification.PayTM.API as API
+import qualified Kernel.External.Notification.PayTM.Types as PayTM
+import Kernel.Tools.Metrics.CoreMetrics (CoreMetrics)
+import Kernel.Types.Common
+import Kernel.Utils.Common
+
+notifyPerson ::
+  ( CoreMetrics m,
+    EncFlow m r,
+    MonadFlow m,
+    ToJSON a
+  ) =>
+  PayTM.PayTMConfig ->
+  PayTM.NotificationReq a ->
+  m ()
+notifyPerson config req = do
+  when (isNothing $ find (\el -> el == req.templateName) config.whitelistedTemplates) $ do
+    logTagInfo tag $ "message not sent as template " <> req.templateName <> " is not whitelisted"
+    pure ()
+  apiKey <- pure . PayTM.getPaytmApiKey =<< decrypt config.apiKey
+  clientId <- pure . PayTM.getPaytmClientId =<< decrypt config.clientId
```

Comment:
`clientId <- PayTM.getPaytmClientId <$> decrypt config.clientId`

By the way this little improvements could be found with hlint:
`hlint . --report`